src/typescript/sdk/src/client/emojicoin-client.ts

```diff
@@ -0,0 +1,332 @@
+    provide: this.provideLiquidity.bind(this),
+    remove: this.removeLiquidity.bind(this),
+  };
+
+  public utils = {
+    emojisToHexStrings: this.emojisToHexStrings.bind(this),
+    emojisToHexSymbol: this.emojisToHexSymbol.bind(this),
+    getEmojicoinInfo: this.getEmojicoinInfo.bind(this),
+    getTransactionEventData: this.getTransactionEventData.bind(this),
+  };
+
+  public rewards = {
+    buy: this.buyWithRewards.bind(this),
+    sell: this.sellWithRewards.bind(this),
+  };
+
+  constructor(args?: { aptos?: Aptos }) {
+    this.aptos = args?.aptos ?? getAptosClient().aptos;
+  }
+
+  async register(
+    registrant: Account,
+    emojis: MarketSymbolEmojis,
+    args: {
+      integrator: AccountAddressInput;
+    },
+    options?: Options
+  ) {
+    const response = await RegisterMarket.submit({
+      aptosConfig: this.aptos.config,
+      registrant,
+      emojis: this.emojisToHexStrings(emojis),
+      integrator: args.integrator,
+      ...options,
+    });
+    const res = this.getTransactionEventData(response);
+    return {
+      ...res,
+      registration: {
+        event: expect(res.events.marketRegistrationEvents.at(0), Expect.Register.Event),
+        model: expect(res.models.marketRegistrationEvents.at(0), Expect.Register.Model),
+      },
+    };
+  }
+
+  async chat(
+    user: Account,
+    emojis: SymbolEmoji[],
+    args: {
+      chatEmojis: AnyEmoji[];
+      emojiIndicesSequence: number[];
+    },
+    options?: Options
+  ) {
+    const chatEmojis = getEmojisInString(args.chatEmojis.join("")) as SymbolEmoji[];
+    const emojiBytes = this.emojisToHexStrings(chatEmojis);
+    const emojiIndicesSequence = new Uint8Array(args.emojiIndicesSequence);
+
+    const response = await Chat.submit({
+      aptosConfig: this.aptos.config,
+      user,
+      emojiBytes,
+      emojiIndicesSequence,
+      ...options,
+      ...this.getEmojicoinInfo(emojis),
+    });
+    const res = this.getTransactionEventData(response);
+    return {
+      ...res,
+      chat: {
+        event: expect(res.events.chatEvents.at(0), Expect.Chat.Event),
+        model: expect(res.models.chatEvents.at(0), Expect.Chat.Model),
+      },
+    };
+  }
+
+  async buy(
+    swapper: Account,
+    emojis: SymbolEmoji[],
+    args: {
+      inputAmount: Uint64;
+      minOutputAmount: Uint64;
+      integrator: AccountAddressInput;
+      integratorFeeRateBPs: number;
+    },
+    options?: Options
+  ) {
+    return await this.swap(swapper, emojis, { ...args, isSell: false }, options);
+  }
+
+  async sell(
+    swapper: Account,
+    emojis: SymbolEmoji[],
+    args: {
+      inputAmount: Uint64;
+      minOutputAmount: Uint64;
+      integrator: AccountAddressInput;
+      integratorFeeRateBPs: number;
+    },
+    options?: Options
+  ) {
+    return await this.swap(swapper, emojis, { ...args, isSell: true }, options);
+  }
+
+  private async swap(
+    swapper: Account,
+    emojis: SymbolEmoji[],
+    args: ExtraSwapArgs,
```

Comment:
For consistency, either put all `args` without a type (i.e. replace this with `args: {...}`), or put all `args` with a dedicated type (i.e. replace `args: {...}` in other functions with `args: Type`).